neuro_san/service/http/server/http_server.py

```diff
@@ -146,11 +147,15 @@ def log_resources_usage(self):
         file descriptors and open inet connections on server port.
         """
         # Get used file descriptors:
-        fds, soft_limit, hard_limit = ServiceResources.get_fd_usage()
-        # Get active TCP connections to our http port:
-        conn = ServiceResources.active_tcp_on_port(self.http_port)
-        self.logger.info({}, "Used: file descriptors %d (%d, %d) connections: %d",
-                         fds, soft_limit, hard_limit, conn)
+        fd_dict, soft_limit, hard_limit = ServiceResources.get_fd_usage()
+        sock_classes = ServiceResources.classify_sockets(self.http_port)
```

Comment:
Does this take awhile to do?
Also: If I recall this happens periodically, but the thing I worry about more is while this logging is going on are there locks within psutil that are going to have everything come to a screeching halt?

Reply:
That's actually a good point if somebody will fancy running these stats every second (we don't allow to run it with <1 second intervals). So I moved the logic out of event loop to be run in a thread alongside.